src/hooks/accounts/useSetActiveAccount.tsx

```diff
@@ -1,59 +1,26 @@
-import { ApolloCache, ApolloClient, NormalizedCacheObject, useApolloClient } from '@apollo/client';
-import { gql } from 'graphql.macro';
-import { find } from 'lodash';
+import { ApolloClient, NormalizedCacheObject } from '@apollo/client';
 import { useCallback } from 'react';
-import { GetAccountsQueryResponse, GET_ACCOUNTS } from './queries/useGetAccountsQuery';
 import { usePersistActiveAccount } from './usePersistActiveAccount';
 
 export const useSetActiveAccount = () => {
-    const [_persistedActiveAccount, setPersistedActiveAccount] = usePersistActiveAccount();
+  const [, setPersistedActiveAccount] = usePersistActiveAccount();
 
-    return useCallback(async (
-        client: ApolloClient<NormalizedCacheObject>,
-        address: string | undefined
+  return useCallback(
+    async (
+      client: ApolloClient<NormalizedCacheObject>,
+      address: string | undefined
     ) => {
-        const accountsData = client.cache?.readQuery<GetAccountsQueryResponse>({
-            query: GET_ACCOUNTS,
-        });
+      setPersistedActiveAccount({
+        id: address,
+      });
 
-        if (!accountsData?.accounts) return;
+      // wait for the local storage changes before finishing the mutation
+      // TODO: find a better way to wait until the local storage changes are
+      // propagated to the resolverRef
+      await new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
This was originally done to ensure that the `GetActiveAccount` query resolver will be already aware of the local storage updates, idk if this is still required - but probably yes.